### PR TITLE
Set app_id in snowflake source connector based on oss/cloud

### LINKED ISSUE #19249: Snowflake source should set application ID in JDBC param
as described in https://airbytehq-team.slack.com/archives/C046LMDDY6N/p1666972075127309

source should set the JDBC URL param application to airbyte_cloud or airbyte_oss

### PR DESCRIPTION
**Pull request by akashkulk**
Set app_id in snowflake source connector based on env (OSS/Cloud)

Closes https://github.com/airbytehq/airbyte/issues/19249

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSourceRunner.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import io.airbyte.integrations.base.adaptive.AdaptiveSourceRunner;
+
+public class SnowflakeSourceRunner {
+
+  public static void main(final String[] args) throws Exception {
+    AdaptiveSourceRunner.baseOnEnv()
+        .withOssSource(() -> new SnowflakeSource(AIRBYTE_OSS))
+        .withCloudSource(() -> new SnowflakeSource(AIRBYTE_CLOUD))
+        .run(args);
+    SCHEDULED_EXECUTOR_SERVICE.shutdownNow();
```
Comment (edgao) [inline airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSourceRunner.java:16]: nice catch, I totally forgot about this on the destination-snowflake PR

**Review by edgao — APPROVED** [review]

**Comment by akashkulk** [comment]
/publish connector=connectors/source-snowflake run-tests=false

**Pull request merged**